pkg/client/client_test.go

```diff
@@ -339,6 +339,106 @@ func TestApplySingle(t *testing.T) {
 			assert.Equal(t, "second-value", configMap.Data["first-param"])
 		})
 	})
+
+	t.Run("updates of ServiceAccount", func(t *testing.T) {
+
+		t.Run("should update service account with last applied configuration", func(t *testing.T) {
```

Comment:
since the "last-appplied-configuration" is used and is set to the same value, then there the SA shouldn't be updated.